Mathlib/Analysis/ODE/PicardLindelof.lean

```diff
@@ -10,397 +10,716 @@ import Mathlib.Topology.MetricSpace.Contracting
+    calc
+      _ ≤ ∫ τ in uIoc t₀.1 t.1, K ^ (n + 1) * |τ - t₀| ^ n / n ! * dist α β := by
+        rw [intervalIntegral.norm_intervalIntegral_eq]
+        apply MeasureTheory.norm_integral_le_of_norm_le (Continuous.integrableOn_uIoc (by fun_prop))
+        apply ae_restrict_mem measurableSet_Ioc |>.mono
+        intro t' ht'
+        -- Can `grind` do this in the future?
+        have ht' : t' ∈ Icc tmin tmax :=
+          subset_trans uIoc_subset_uIcc (uIcc_subset_Icc t₀.2 t.2) ht'
+        rw [← dist_eq_norm, compProj_of_mem, compProj_of_mem]
+        exact dist_comp_iterate_next_le hf hx _ ⟨t', ht'⟩ (hn _)
+      _ ≤ (K * |t.1 - t₀.1|) ^ (n + 1) / (n + 1) ! * dist α β := by
+        apply le_of_abs_le
+        -- critical: `integral_pow_abs_sub_uIoc`
+        rw [← intervalIntegral.abs_intervalIntegral_eq, intervalIntegral.integral_mul_const,
+          intervalIntegral.integral_div, intervalIntegral.integral_const_mul, abs_mul, abs_div,
+          abs_mul, intervalIntegral.abs_intervalIntegral_eq, integral_pow_abs_sub_uIoc, abs_div,
+          abs_pow, abs_pow, abs_dist, NNReal.abs_eq, abs_abs, mul_div, div_div, ← abs_mul,
+          ← Nat.cast_succ, ← Nat.cast_mul, ← Nat.factorial_succ, Nat.abs_cast, ← mul_pow]
+
+/-- The `n`-th iterate of `next` is Lipschitz continuous with respect to `FunSpace`, with constant
+$(K \max(t_{\mathrm{max}}, t_{\mathrm{min}})^n / n!$. -/
+lemma dist_iterate_next_iterate_next_le (hf : IsPicardLindelof f t₀ x₀ a r L K)
+    (hx : x ∈ closedBall x₀ r) (α β : FunSpace t₀ x₀ r L) (n : ℕ) :
+    dist ((next hf hx)^[n] α) ((next hf hx)^[n] β) ≤
+      (K * max (tmax - t₀) (t₀ - tmin)) ^ n / n ! * dist α β := by
+  rw [← MetricSpace.isometry_induced FunSpace.toContinuousMap FunSpace.toContinuousMap.injective
+    |>.dist_eq, ContinuousMap.dist_le]
+  · intro t
+    apply le_trans <| dist_iterate_next_apply_le hf hx α β n t
+    gcongr
+    exact abs_sub_le_max_sub t.2.1 t.2.2 _
+  · have : 0 ≤ max (tmax - t₀) (t₀ - tmin) := le_max_of_le_left <| sub_nonneg_of_le t₀.2.2
+    positivity
+
+/-- Some `n`-th iterate of `next` is a contracting map, and its associated Lipschitz constant is
+independent of the initial point. -/
+lemma exists_contractingWith_iterate_next (hf : IsPicardLindelof f t₀ x₀ a r L K) :
+    ∃ (n : ℕ) (C : ℝ≥0), ∀ (x : E) (hx : x ∈ closedBall x₀ r),
+      ContractingWith C (next hf hx)^[n] := by
+  obtain ⟨n, hn⟩ := FloorSemiring.tendsto_pow_div_factorial_atTop (K * max (tmax - t₀) (t₀ - tmin))
+    |>.eventually (gt_mem_nhds zero_lt_one) |>.exists
+  have : (0 : ℝ) ≤ (K * max (tmax - t₀) (t₀ - tmin)) ^ n / n ! := by
+    have : 0 ≤ max (tmax - t₀) (t₀ - tmin) := le_max_of_le_left <| sub_nonneg_of_le t₀.2.2
+    positivity
+  refine ⟨n, ⟨_, this⟩, fun x hx ↦ ?_⟩
+  exact ⟨hn, LipschitzWith.of_dist_le_mul fun α β ↦ dist_iterate_next_iterate_next_le hf hx α β n⟩
+
+/-- The map `next` has a fixed point in the space of curves. This will be used to construct a
```

Comment:
The following lines are OK, but it seems that an elementary API is missing here, that would handle maps which have a contracting iterate and provide their fixed point.